consensus/taiko/consensus.go

```diff
@@ -229,12 +229,14 @@ func (t *Taiko) FinalizeAndAssemble(chain consensus.ChainHeaderReader, header *t
 
 	// Verify anchor transaction
 	if len(body.Transactions) != 0 { // Transactions list might be empty when building empty payload.
-		isAnchor, err := t.ValidateAnchorTx(body.Transactions[0], header)
-		if err != nil {
-			return nil, err
-		}
-		if !isAnchor {
-			return nil, ErrAnchorTxNotFound
+		if body.Transactions[0].IsAnchor() {
```

Comment:
I remembered that the function `IsAnchor()` seems always to return `false`?
And I think using `ValidateAnchorTx` is enough.

Reply:
It's ok if the first tx is anchor tx.

Reply:
It can't happen that the first tx in the block to be finalised is not an anchor tx, can it?

Reply:
A new change in this [commit](https://github.com/taikoxyz/taiko-geth/pull/357/commits/6509066d2f8dcf246f41fc555776fa9328ed3f2a).